packages/next-app/src/app/stableswapsimulator/(components)/VariantDataForm.tsx

```diff
@@ -0,0 +1,86 @@
+"use client";
+
+import { StableSwapSimulatorDataSchema } from "@balancer-pool-metadata/schema";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+
+import { Input } from "#/components/Input";
+import { Select, SelectItem } from "#/components/Select";
+import { AnalysisData, useStableSwap } from "#/contexts/StableSwapContext";
+
+export default function NewDataForm() {
+  const {
+    baselineData,
+    variantData,
+    setVariantData,
+    setIndexAnalysisToken,
+    indexAnalysisToken,
+  } = useStableSwap();
+
+  const {
+    register,
+    setValue,
+    getValues,
+    formState: { errors },
+  } = useForm<typeof StableSwapSimulatorDataSchema._type>({
+    resolver: zodResolver(StableSwapSimulatorDataSchema),
+    mode: "onChange",
+  });
+
+  const handleChange = () => {
+    if (Object.keys(errors).length) return;
+    const data = getValues();
+    setVariantData(data as AnalysisData);
+  };
+
+  useEffect(() => {
+    if (variantData?.swapFee) setValue("swapFee", variantData?.swapFee);
+    if (variantData?.ampFactor) setValue("ampFactor", variantData?.ampFactor);
+  }, [variantData]);
```

Comment:
Not sure I follow this, can you elaborate? If you want to set a default value you can add that to the `useForm` hook